onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperator.cpp

```diff
@@ -111,6 +111,38 @@ namespace Dml
         }
     }
 
+    void DmlOperator::SetDmlOperatorDesc(
+        const MLOperatorGraphDesc& operatorGraphDesc,
+        const MLOperatorKernelCreationContext& kernelInfo
+        )
+    {
+        // Initialize should only be called once.
+        assert(m_compiledOperator == nullptr);
+
+        // DML doesn't support empty tensors. If an operator is still executable with empty tensors, the empty tensors
+        // should be removed or massaged depending on the definition.
+        for (const TensorDesc& desc : m_inputTensorDescs)
+        {
+            if (OperatorHelper::ContainsEmptyDimensions(desc.GetSizes()))
+            {
+                return;
+            }
+        }
+
+        for (const TensorDesc& desc : m_outputTensorDescs)
+        {
+            if (OperatorHelper::ContainsEmptyDimensions(desc.GetSizes()))
+            {
+                return;
+            }
+        }
+
+        ComPtr<IMLOperatorKernelCreationContextPrivate> contextPrivate;
+        ORT_THROW_IF_FAILED(kernelInfo.GetInterface()->QueryInterface(contextPrivate.GetAddressOf()));
+        assert(contextPrivate->IsDmlGraphNode());
```

Comment:
Nothing guarantees this is the case.  This should need a fallback similar to the method above.  Please also add a conformance test that will hit this by using free dimensions

Reply:
I had added a comment on the signature of this method that currently it is only supported in the context of dml graph because making it work for non-graph route is not very trivial right now and I didn't want to overload this PR too much.

But I will again look into this if it is possible trivially.